### PR TITLE
ci: Create GitHub workflow

### PR DESCRIPTION
**Pull request by yuhao900914**

### CODE REVIEW

**Comment by kelsonpw** [comment]
Looks like the Publish to Github Pages failed

**Review by jooohhn — CHANGES_REQUESTED** [review]
LGTM, just a couple of points

File: release.config.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      '@semantic-release/npm',
+      {
+        npmPublish: true,
+      },
+    ],
+    ['@semantic-release/github'],
+    [
+      '@semantic-release/git',
+      {
+        assets: ['src/index.ts', 'docs', 'package.json'],
```
Comment (jooohhn) [inline release.config.js:29]: Does src/index.ts need to be managed by the semantic-release bot?

Comment (kelsonpw) [reply release.config.js:29]: we need to make sure that we only ship the `lib` directory which is built after running `yarn prepare` or `bob build`.  and I dont think this current PR is accomplishing that.

**Review by kelsonpw — CHANGES_REQUESTED** [review]
We need to make sure we only ship the lib directory and no other assets

File: release.config.js (hunk 1 above)
Comment (yuhao900914) [reply release.config.js:29]: the `file` parameter in the Package.json determine which file we pushed to npm. Currently, I set "files": ["lib/**"], which means we only ship the files under the ilb directory.

Comment (yuhao900914) [reply release.config.js:29]: @kelsonpw 
Reference:https://github.com/semantic-release/npm/issues/128
And I saw the previously the files setting is as the following, which not including only the lib directory, which one should we use?

Comment (kelsonpw) [reply release.config.js:29]: We want to follow that same list, you are right.  Thanks for catching that.  I think john is right about removing 'src/index.ts' from assets though

**Comment by kelsonpw** [comment]
@yuhao900914 you are correct about the files to ship.  Here is what is included when downloading the current version:

image

So we want to publish the same output with this new workflow.

**Review by kelsonpw — APPROVED** [review]

**Pull request merged**